include/h2o/http2_internal.h

```diff
@@ -224,6 +232,7 @@ struct st_h2o_http2_conn_t {
     } _write;
     h2o_cache_t *push_memo;
     h2o_http2_casper_t *casper;
+    int _sent_goaway;
```

Comment:
I believe `conn->state == H2O_HTTP2_CONN_STATE_HALF_CLOSED` would be sufficient to see if a GOAWAY frame has been sent. Please refer to the comments on `h2o_http2_conn_state_t`.